ee/hogai/graph/agent_modes/nodes.py

```diff
@@ -465,6 +466,37 @@ async def arun(self, state: AssistantState, config: RunnableConfig) -> PartialAs
                 raise ValueError(
                     f"Tool '{tool_call.name}' returned {type(result).__name__}, expected LangchainToolMessage"
                 )
+        except MaxToolError as e:
```

Comment:
I think this whole bit after the except can be wrapped in a function or `MaxToolErrorHandler` class, so we can extend it with more errors? or even better, the error has a `to_string` function that wraps this (you can make a mixin for all MaxTool error classes)